src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerRequestForInformationJoint.java

```diff
@@ -0,0 +1,119 @@
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.api.CCDConfig;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.ccd.sdk.api.ConfigBuilder;
+import uk.gov.hmcts.ccd.sdk.api.callback.AboutToStartOrSubmitResponse;
+import uk.gov.hmcts.divorce.caseworker.service.notification.RequestForInformationNotification;
+import uk.gov.hmcts.divorce.common.ccd.PageBuilder;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.RequestForInformation;
+import uk.gov.hmcts.divorce.divorcecase.model.RequestForInformationList;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.model.UserRole;
+import uk.gov.hmcts.divorce.notification.NotificationDispatcher;
+import uk.gov.hmcts.divorce.notification.exception.NotificationTemplateException;
+
+import java.util.Collections;
+import java.util.List;
+
+import static uk.gov.hmcts.divorce.divorcecase.model.State.AwaitingDocuments;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CASE_WORKER;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CITIZEN;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.JUDGE;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.LEGAL_ADVISOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SOLICITOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SUPER_USER;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE;
+
+@Slf4j
+@RequiredArgsConstructor
+@Component
+public class CaseworkerRequestForInformationJoint implements CCDConfig<CaseData, State, UserRole> {
+
+    public static final String CASEWORKER_REQUEST_FOR_INFORMATION_JOINT = "caseworker-request-for-information-joint";
+
+    public static final String REQUEST_FOR_INFORMATION_NOTIFICATION_FAILED_ERROR
+        = "Unable to send Request for Information Notification for Case Id: ";
+
+    private final CaseworkerRequestForInformationHelper helper;
+
+    private final RequestForInformationNotification requestForInformationNotification;
+
+    private final NotificationDispatcher notificationDispatcher;
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
+            .event(CASEWORKER_REQUEST_FOR_INFORMATION_JOINT)
+            .forAllStates()
+            .name("Request For Information")
+            .description("Request for information")
+            .showCondition("applicationType=\"jointApplication\"")
+            .showSummary()
+            .showEventNotes()
+            .endButtonLabel("Submit")
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .grant(CREATE_READ_UPDATE, CASE_WORKER)
+            .grantHistoryOnly(SUPER_USER, LEGAL_ADVISOR, JUDGE, SOLICITOR, CITIZEN, JUDGE))
```

Comment:
I believe citizen cannot view history via XUI so may be remove it from history list?

Reply:
Fixed